projects/igniteui-angular/src/lib/tree/tree-node/tree-node.component.ts

```diff
@@ -0,0 +1,355 @@
+    private childrenContainer: ElementRef;
+
+    // TODO: this should probably be dropped from the API
+    /**
+     * The unique ID of the node
+     */
+    public id = `igxTreeNode_${nodeId++}`;
+
+    constructor(
+        @Inject(IGX_TREE_COMPONENT) public tree: IgxTree,
+        protected selectionService: IgxTreeSelectionService,
+        protected treeService: IgxTreeService,
+        protected cdr: ChangeDetectorRef,
+        protected builder: AnimationBuilder,
+        private element: ElementRef<HTMLElement>,
+        @Optional() @SkipSelf() @Inject(IGX_TREE_NODE_COMPONENT) public parentNode: IgxTreeNode<any>
+    ) {
+        super(builder);
+    }
+
+    /**
+     * @hidden @internal
+     */
+    public get showSelectors() {
+        return this.tree.selection !== IGX_TREE_SELECTION_TYPE.None;
+    }
+
+    /**
+     * @hidden @internal
+     */
+    public get indeterminate(): boolean {
+        return this.selectionService.isNodeIndeterminate(this);
+    }
+
+    /** The depth of the node, relative to the root
+     *
+     * ```html
+     * <igx-tree>
+     *  ...
+     *  <igx-tree-node #node>
+     *      My level is {{ node.level }}
+     *  </igx-tree-node>
+     * </igx-tree>
+     * ```
+     *
+     * ```typescript
+     * const node: IgxTreeNode<any> = this.tree.findNodes(data[12])[0];
+     * const level: number = node.level;
+     * ```
+     */
+    public get level(): number {
+        return this.parentNode ? this.parentNode.level + 1 : 0;
+    }
+
+    /** Get/set whether the node is selected. Supporst two-way binding.
+     *
+     * ```html
+     * <igx-tree>
+     *  ...
+     *  <igx-tree-node *ngFor="let node of data" [(selected)]="node.selected">
+     *      {{ node.label }}
+     *  </igx-tree-node>
+     * </igx-tree>
+     * ```
+     *
+     * ```typescript
+     * const node: IgxTreeNode<any> = this.tree.findNodes(data[0])[0];
+     * const selected = node.selected;
+     * node.selected = true;
+     * ```
+     */
+    @Input()
+    public get selected(): boolean {
+        return this.selectionService.isNodeSelected(this);
+    }
+
+    public set selected(val: boolean) {
+        if (val && !this.selectionService.isNodeSelected(this)) {
+            this.selectionService.selectNodesWithNoEvent([this]);
+        }
+        if (!val && this.selectionService.isNodeSelected(this)) {
+            this.selectionService.deselectNodesWithNoEvent([this]);
+        }
+    }
+
+    /** Get/set whether the node is expanded
+     *
+     * ```html
+     * <igx-tree>
+     *  ...
+     *  <igx-tree-node *ngFor="let node of data" [expanded]="node.name === this.expandedNode">
+     *      {{ node.label }}
+     *  </igx-tree-node>
+     * </igx-tree>
+     * ```
+     *
+     * ```typescript
+     * const node: IgxTreeNode<any> = this.tree.findNodes(data[0])[0];
+     * const expanded = node.expanded;
+     * node.expanded = true;
+     * ```
+     */
+    @Input()
+    public get expanded() {
```

Comment:
Also you need `expandedChange` for `[(expanded)]` binding.

Reply:
My mistake, I've added that